pkg/controller/cincinnati/cincinnati_controller_test.go

```diff
@@ -0,0 +1,140 @@
+package cincinnati
+
+import (
+	"os"
+	"testing"
+
+	imagev1 "github.com/openshift/api/image/v1"
+	"github.com/openshift/cincinnati-operator/pkg/apis"
+	cv1alpha1 "github.com/openshift/cincinnati-operator/pkg/apis/cincinnati/v1alpha1"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	"github.com/stretchr/testify/assert"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	testName                 = "foo"
+	testNamespace            = "bar"
+	testCincinnatiKind       = "testKind"
+	testCincinnatiAPIVersion = "testAPIVersion"
+	testOperandImage         = "testOperandImage"
+	testReplicas             = -1
+	testRegistry             = "testRegistry"
+	testRepository           = "testRepository"
+	testGitHubOrg            = "testGitHubOrg"
+	testGitHubRepo           = "testGitHubRepo"
+	testGitHubBranch         = "testGitHubBranch"
+)
+
+var _ reconcile.Reconciler = &ReconcileCincinnati{}
+
+func TestMain(m *testing.M) {
+	if err := apis.AddToScheme(scheme.Scheme); err != nil {
+		log.Error(err, "Failed adding apis to scheme")
+		os.Exit(1)
+	}
+	if err := imagev1.AddToScheme(scheme.Scheme); err != nil {
+		log.Error(err, "Failed adding apis to scheme")
+		os.Exit(1)
+	}
+
+	exitVal := m.Run()
+	os.Exit(exitVal)
+}
+
+// TestReconcileComplete ensures that the cincinnati object is reconciled successfully
+func TestReconcileComplete(t *testing.T) {
+	cincinnati := newDefaultCincinnati()
+	r := newTestReconciler(cincinnati)
+	request := newRequest(cincinnati)
+
+	_, err := r.Reconcile(request)
+	if err != nil {
+		t.Fatal(err)
+	}
+	instance := &cv1alpha1.Cincinnati{}
+	r.client.Get(nil, request.NamespacedName, instance)
```

Comment:
Do not pass a nil Context, even if a function permits it; pass `context.TODO`

Reply:
Sounds like a sensible default.